examples/integration/integration_test.go

```diff
@@ -1466,3 +1467,58 @@ func testResponseStrings(t *testing.T, port int) {
 	}
 
 }
+
+func TestRequestQueryParams(t *testing.T) {
+	testRequestQueryParamsKind(t, 8080, "GET")
+	testRequestQueryParamsKind(t, 8080, "POST")
+	testRequestQueryParamsKind(t, 8080, "PostForm")
+}
```

Comment:
Just a matter of style, instead of doing this, have you considered writing subtests?

```go
func TestRequestQueryParams(t *testing.T) {
	t.Run("GET", func(t *testing.T) {
		...
	})
	t.Run("POST", func(t *testing.T) {
		...
	})
	...
}
```

Reply:
I see other examples are written in this style.

Reply:
I don't quite agree. The way that you're passing in a string to switch on inside the function is pretty nasty. I would suggest separating out the shared content into a function, if at all. Just having 3 different tests would be fine too.